### PR TITLE
build(lint): add `eslint-plugin-react-hooks`

### PR DESCRIPTION
**Pull request by SimonFinney**
## Affected issue

Contributes to #808 

## Proposed change

Add `eslint-plugin-react-hooks` package and configuration

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "plugin:import/recommended",
     "plugin:jsx-a11y/recommended",
     "plugin:react/recommended",
+    "plugin:react-hooks/recommended",
```
Comment (diego-codes) [inline .eslintrc:9]: Do you need to include it as a plugin to work?

Comment (SimonFinney) [reply .eslintrc:9]: It should be enough to extend the configuration unless we need to customize the rules further - https://www.npmjs.com/package/eslint-plugin-react-hooks#installation

Comment (diego-codes) [reply .eslintrc:9]: Gotcha, perfect! I'll approve then.

**Review by diego-codes — APPROVED** [review]

**Pull request merged**